### ISSUE TITLE
ModelCheckpoint as training checkpointing

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#186]
I would like to use `ModelCheckpoint` to store model, optimizer, lr_scheduler like this:
```python
training_saver = ModelCheckpoint("checkpoint",
                                 filename_prefix="checkpoint",
                                 save_interval=1000,  # Save every 1000 iterations
                                 n_saved=1,
                                 atomic=True,
                                 save_as_state_dict=True,
                                 create_dir=True)
trainer.add_event_handler(Events.ITERATION_COMPLETED, 
                          training_saver, 
                          {
                              "model": model,
                              "optimizer": optimizer,
                              "lr_scheduler": lr_scheduler
                          })
```
before engine checkpointing issue is resolved.

However, if I use `save_as_state_dict=True` the [following check](https://github.com/pytorch/ignite/blob/0747d4817ebbd78302c12c3ebe814ce7b34572c3/ignite/handlers/checkpoint.py#L142) won't allow to save `optimizer` and `lr_scheduler`. 
Maybe we can replace it by just checking if object has attribute `state_dict()`. 

What do you think about this ? 
Thanks

**Comment by alykhantejani** [comment]
Yeah, that makes sense - especially as the arg only mentions state_dicts. 

I'm in favor of this change

**Comment by vfdev-5** [comment]
Fix by #187
